### PR TITLE
Feature/profile description field

### LINKED ISSUE #117: Add description field to tutor profile

### PR DESCRIPTION
**Pull request by alexs-rufer**
## PR Description:

I added a new `description` field for tutor profiles (suggested by our coaches), allowing tutors to share their experience or explain why they would be great in a tutoring role. In addition, the tutor profile creation flow has been updated to include a corresponding text box for this new field, and associated tests have been implemented.

## Files changed

- **Profile.kt**
  - Added a new `description` field to the `Profile` data class with a default empty string value. This enables tutors to describe their experience. The default value as empty string is used because student are not supposed to have one.

- **ProfilesRepositoryFirestore.kt**
  - Updated `documentToProfile` to correctly retrieve the `description` field from Firestore documents.
  - Also updated `profileToMap` function  to include the new `description` field.

- **CreateTutorProfile.kt**
  - Added an `OutlinedTextField` to capture tutor experience, labeled "Do you have any experience as a tutor?". This field displays placeholder text to guide tutors on what to enter if they have no formal experience. I set a maximum of 4 lines of description, which seemed enough to me but could be debated. Tell me what you think.

- **CreateTutorProfileTest.kt**
  - Added a new test, `descriptionField_updatesCorrectly`, to verify that the description field is displayed and updates as expected when text is entered.
  - Refactored the profile setup process across tests by adding a `getMockProfile` function, because we were creating like 4 times the same profile :)

**Minor Updates**
- Updated **AddLessonTest.kt** and **OverViewTest.kt** to ensure profiles created through the Profile constructor align with the new `description` field in the `Profile` data class structure. (The app was failing to build without that description field explicitly added, even with the default value of that field..)

## Screenshots 
Here is a screenshot of the added text field in the createTutor screen. Nothing crazy.. :)

Screenshot 2024-11-12 at 21 15 54

Screenshot 2024-11-12 at 21 16 04

Close #117

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/project/ui/profile/CreateTutorProfileTest.kt [hunk 1]
```diff
@@ -178,4 +162,22 @@ class CreateTutorProfileTest {
...
+    composeTestRule
+        .onNodeWithTag("experienceField")
+        .assertTextContains("Do you have any experience as a tutor ?")
+
+    val descriptionText = "I have experience teaching math and physics."
+    composeTestRule.onNodeWithTag("experienceField").performTextInput(descriptionText)
+
+    composeTestRule.onNodeWithTag("experienceField").assertTextContains(descriptionText)
+  }
 }
```
Comment (aferrante1) [inline app/src/androidTest/java/com/github/se/project/ui/profile/CreateTutorProfileTest.kt:183]: Great work! The test effectively checks if the description field updates correctly and handles the state well.

File: app/src/androidTest/java/com/github/se/project/ui/profile/CreateTutorProfileTest.kt [hunk 2]
```diff
@@ -38,21 +40,24 @@ class CreateTutorProfileTest {
...
+          firstName = "First",
+          lastName = "Last",
+          phoneNumber = "1234567890",
+          role = Role.TUTOR,
+          section = Section.GM,
+          academicLevel = AcademicLevel.MA2,
+          languages = listOf(),
+          subjects = listOf(),
+          schedule = listOf())
+
```
Comment (aferrante1) [inline app/src/androidTest/java/com/github/se/project/ui/profile/CreateTutorProfileTest.kt:57]: Great idea to define the `MockProfile` outside of the test, as it helps eliminate repetition and keeps the test code cleaner.

File: app/src/main/java/com/github/se/project/ui/profile/CreateTutorProfile.kt [hunk 3]
```diff
@@ -87,6 +88,20 @@ fun CreateTutorProfile(
...
+                  modifier =
+                      Modifier.fillMaxWidth().padding(vertical = 4.dp).testTag("experienceField"),
+                  label = { Text("Do you have any experience as a tutor ?") },
+                  placeholder = {
+                    Text(
+                        "Share your previous tutoring experience, or if you don't have any, explain why you would be a great tutor.")
+                  },
+                  shape = MaterialTheme.shapes.medium,
+                  maxLines = 4)
             }
```
Comment (aferrante1) [inline app/src/main/java/com/github/se/project/ui/profile/CreateTutorProfile.kt:105]: The use of the OutlinedTextField for the experience input is clear and intuitive. The maxLines Property to 4 is a good idea to limit the space for input.

**Review by aferrante1 — COMMENTED** [review]
Great and meaningful addition to the user profile, just check the CI and we are good to merge!

**Review by LeaBlancher — APPROVED** [review]
This PR was already reviewed by Alessandro before you merged the main branch into it. I agree with all coments Alessandro wrote, you PR is really great and it a good improve that we add a description for the tutor profiles, it will be very useful for user to choose between tutors. Good job!
This PR is ready to be merged.

File: app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt [hunk 4]
```diff
@@ -75,17 +75,17 @@ class TutorEndToEndTest {
...
+          googleUid = "mockTutor",
+          firstName = "Ozymandias",
+          lastName = "Halifax",
+          phoneNumber = "1234567890",
+          role = Role.STUDENT,
+          section = Section.IN,
+          academicLevel = AcademicLevel.BA3,
+          languages = listOf(Language.ENGLISH),
+          subjects = listOf(Subject.AICC),
+          schedule = List(7) { List(12) { 0 } })
```
[the remark this answers is not in the document]
Comment (A-M-00) [reply app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt:89]: Agreed!

**Comment by A-M-00** [comment]
Looks good to me

**Review by A-M-00 — APPROVED** [review]
Looks good to me! Just approving the last commit

**Pull request merged**